### PR TITLE
Add timestamps to items on upsert

### PR DESCRIPTION
**Pull request by jde**
Add timestamps to items on Upsert operations: #253

### CODE REVIEW

File: internal/sponge/item/item.go [hunk 1]
```diff
@@ -33,6 +34,14 @@ func Upsert(context interface{}, db *db.DB, item *Item) error {
 		return err
 	}
 
+	// If CreatedAt is not set, set it. This allows for CreatedAt to be set for data import.
+	if item.CreatedAt.IsZero() {
```
Comment (wyattjoh) [inline internal/sponge/item/item.go]: I'd rather see this being added above where we check to see if the ID is set. If the ID is not set, then we obviously need to set the `CreatedAt` timestamp.

File: internal/sponge/item/model.go [hunk 2]
```diff
@@ -27,10 +28,12 @@ const sourceIDField = "id"
...
-	ID      string                 `bson:"item_id" json:"item_id" validate:"required,min=1"`
-	Type    string                 `bson:"type" json:"type" validate:"required,min=2"`
-	Version int                    `bson:"version" json:"version" validate:"required,min=1"`
-	Data    map[string]interface{} `bson:"data" json:"data"`
+	ID        string                 `bson:"item_id" json:"item_id" validate:"required,min=1"`
+	Type      string                 `bson:"type" json:"type" validate:"required,min=2"`
+	Version   int                    `bson:"version" json:"version" validate:"required,min=1"`
+	Data      map[string]interface{} `bson:"data" json:"data"`
+	CreatedAt time.Time              `bson:"created_at" json:"created_at"`
+	UpdatedAt time.Time              `bson:"updated_at,omitempty" json:"updated_at,omitempty"`
```
Comment (wyattjoh) [inline internal/sponge/item/model.go]: I don't think that the `omitempty` should be there. If a document is created, it was also "updated" at that time, so for each `Item`, we always have a `created_at` and `updated_at` field available.

File: internal/sponge/item/item.go (hunk 1 above)
Comment (jde) [inline internal/sponge/item/item.go]: The creation of an id is decoupled from setting CreatedAt in cases where items are being imported as data and the ids are being inferred. Also, there are times when a frontend will want to Create a new item but specify it's ID, such as creating a "singleton" item that can be identified by a given key, such as a default thread, config object, etc...

File: internal/sponge/item/model.go (hunk 2 above)
Comment (jde) [inline internal/sponge/item/model.go]: Updated.

File: internal/sponge/item/item.go (hunk 1 above)
Comment (jde) [inline internal/sponge/item/item.go]: I clarified this in the comments here: 950f874

**Pull request merged**